src/main/java/picocli/CommandTokenizer.java

```diff
@@ -0,0 +1,371 @@
+package picocli;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+
+public class CommandTokenizer {
```

Comment:
Would be great to have some javadoc that describes what this class does, ideally with an example of how it can be used.

Reply:
Good idea. I'm waiting until it's done though, in case we decide anything needs to be changed.